src/testrun.rs

```diff
@@ -46,63 +48,145 @@ impl Display for Outcome {
     }
 }
 
-#[derive(Clone, Debug)]
+static FRAMEWORKS: LazyLock<Vec<(&str, Framework)>> = LazyLock::new(|| {
+    vec![
+        ("pytest", Framework::Pytest),
+        ("vitest", Framework::Vitest),
+        ("jest", Framework::Jest),
+        ("phpunit", Framework::PHPUnit),
+    ]
+});
+static EXTENSIONS: LazyLock<Vec<(&str, Framework)>> =
+    LazyLock::new(|| vec![(".py", Framework::Pytest), (".php", Framework::PHPUnit)]);
+
+fn check_substring_before_word_boundary(string: &str, substring: &str) -> bool {
+    if let Some((_, suffix)) = string.to_lowercase().split_once(substring) {
+        match suffix.chars().next() {
+            None => {
+                return true;
+            }
+            Some(first_char) => {
+                if !first_char.is_alphanumeric() {
+                    return true;
+                }
+            }
+        }
+    }
+    false
+}
+
+pub fn check_testsuites_name(testsuites_name: &str) -> Option<Framework> {
+    for (name, framework) in &*FRAMEWORKS {
+        if check_substring_before_word_boundary(testsuites_name, name) {
+            return Some(framework.to_owned());
+        }
+    }
+    None
```

Comment:
```suggestion
    FRAMEWORKS.iter().filter_map(|(name, framework)| {
        check_substring_before_word_boundary(testsuites_name, name).then_some(*framework)
    }).next()
```

a bit more functional combinators :-D